src/expr.rs

```diff
@@ -674,9 +674,13 @@ impl<'a> Rewrite for Loop<'a> {
             ControlBraceStyle::AlwaysNextLine => alt_block_sep.as_str(),
             ControlBraceStyle::AlwaysSameLine => " ",
         };
+        let block_width = try_opt!(width.checked_sub(label_string.len() + self.keyword.len() +
+                                                     extra_offset(&pat_expr_string, inner_offset) +
+                                                     1));
```

Comment:
Yes you are. Width passed to Block::rewrite is only used for the empty loop case (and for single line unsafe block, which doesn't apply here). In the non-empty case, width is reset to max width minus offset.